### DISCUSSION TITLE
Plausible can't setup the clickhouse in my mac

### DISCUSSION THREAD

**Discussion by Lalarukh-khan** [plausible/analytics#2577] [Self-Hosted Support] [answered]
Hi!

I'm trying to run my project, but it can't have a connectivity with the clickhouse. However it has finally generated the DB in my postgres server named "plausible_dev". But can't have a connection with the clickhouse.
I'm getting this respose in terminal.
`[error] Clickhousex.Protocol (#PID<0.889.0>) failed to connect: ** (ErlangError) Erlang error: :econnrefused`
I have thoroughly read the docs, but found no resource for this bug. 
Seeking Guidance!
Thanks
![Screenshot 2023-01-10 at 2 43 32 PM](https://user-images.githubusercontent.com/72685841/211527326-3489e516-eec6-484a-a020-8d13484a3949.png)

**Comment by ruslandoga** [comment, marked as answer]
@Lalarukh-khan 👋

The error message means that ClickHouse is not reachable at the [URL provided in the configuration.](https://github.com/plausible/analytics/blob/256e8263c02c1d5b2b86a22cd6808029b09ac221/config/runtime.exs#L77-L82)

Could you please share the URL you use for ClickHouse? How do you run ClickHouse?

**Reply by Lalarukh-khan** [reply]
This is the code related to the clickhouse URL developed inside:
`ch_db_url =
  get_var_from_path_or_env(
    config_dir,
    "CLICKHOUSE_DATABASE_URL",
    "http://plausible_events_db:8123/plausible_events_db"
  )`

**Reply by ruslandoga** [reply]
Yes, and if you haven't set `CLICKHOUSE_DATABASE_URL`, Plausible would try to connect to ClickHouse at `http://plausible_events_db:8123/plausible_events_db`. This setup assumes [`docker-compose`](https://github.com/plausible/hosting/blob/5097a507fc68b2e7fd8eb59ba08ec16f87fde440/docker-compose.yml#L16-L26) deployment.

**Reply by Lalarukh-khan** [reply]
Then how can I call the URL. I mean which url is suppose to be written here? Thanks

**Reply by ruslandoga** [reply]
The URL pointing at your ClickHouse instance. More info on how to run Plausible, PostgreSQL, ClickHouse and other services using Docker Compose is [here.](https://plausible.io/docs/self-hosting)
